okhttp/src/test/java/okhttp3/OkHttpTest.kt

```diff
@@ -16,11 +16,45 @@
 package okhttp3
 
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.Assert.assertEquals
+import org.junit.Assert.fail
 import org.junit.Test
 
 class OkHttpTest {
   @Test
   fun testVersion() {
     assertThat(OkHttp.VERSION).matches("[0-9]+\\.[0-9]+\\.[0-9]+(-.+)?")
   }
+
+  @Test
+  fun testReleaseVersionCompatibity() {
+    OkHttp.checkVersion(module = "okhttp-mockwebserver", moduleVersion = "4.20.0", okhttpVersion = "4.20.0")
+  }
+
+  @Test
+  fun testReleaseVersionIncompatibity() {
+    try {
+      OkHttp.checkVersion(
+          module = "mockwebserver", moduleVersion = "4.20.0", okhttpVersion = "4.21.0"
+      )
+      fail()
+    } catch (ise: IllegalStateException) {
+      assertEquals("com.squareup.okhttp3:okhttp:4.21.0 is not compatible with com.squareup.okhttp3:mockwebserver:4.20.0", ise.message)
+    }
+  }
+
+  @Test
+  fun testSnapshotVersionCompatibity() {
+    OkHttp.checkVersion(module = "okhttp-mockwebserver", moduleVersion = "4.20.0-SNAPSHOT", okhttpVersion = "4.20.0-SNAPSHOT")
+  }
+
+  @Test
+  fun testSnapshotReleaseVersionIncompatibity() {
+    OkHttp.checkVersion(module = "okhttp-mockwebserver", moduleVersion = "4.20.0", okhttpVersion = "4.21.0-SNAPSHOT")
```

Comment:
what makes this case okay?

Reply:
Basically to not be annoying to anyone that is testing a dev version of OkHttp.  Probably testing a PR build, or possibly knowingly releases with locally patched lib that we don't control.

But if you are using released tagged versions of your libs, then you don't cross the versions.

![image](https://user-images.githubusercontent.com/231923/89449964-b8bdf100-d751-11ea-88c0-c6c222d3556d.png)